### DISCUSSION TITLE
Any client for OpenSearch 2.x?

### DISCUSSION THREAD

**Discussion by sorin-costea** [quarkusio/quarkus#29517] [Q&A] [answered]
I just upgraded my test AWS cluster to OpenSearch 2.3 and lo, errors everywhere. Not surprising, but it's surprising that I can't find any Quarkus discussions/issues/whatever about OS 2.x. I don't use Hibernate just the REST high level client, which I would migrate to whatever else - if there was anything in the Quarkus universe speaking OpenSearch 2.x...

**Comment by maxandersen** [comment]
The elasticsearch highlevel Client became problematic when they changed license and became non-opensource. That's why things started moving slow. 

Now; they cleaned up things and things are now looking better.

 Cc @loicmathieu as he could be able to put some light on options available now or later.

**Comment by loicmathieu** [comment, marked as answer]
Hi,
I was contacted a few days ago about an OpenSearch extension, and we already discuss it a few month ago.
Yes, an OpenSearch extension will be great in the Quarkiverse now that the two starts to diverge.
Someone is willing to contribute it and I will help him if needed, I asked him to open a disussion in Zulip, @sorin-costea I'll ping you when he starts the discussion.

**Reply by apatrida** [reply]
See: https://github.com/sort-dev/quarkus-opensearch
as starting point, and go back to the commit for last 2.14.x Quarkus version 75645a90f4b0b37cfcd5bede60085fb13acf09d3 as the best starting point.

**Comment by apatrida** [comment]
Someone can start with:

https://github.com/sort-dev/quarkus-opensearch

and go back from there to the 0.3.0 tag of the codebase as it is 2.14.x or so of Quarkus (and the 0.5.0 isn't quite there yet for Quarkus 3.0.0-Alpha)

It was based on the ES extension but with some changes to work better with the test containers from opensearch and other minor changes.  A starting point at least for someone.

**Comment by loicmathieu** [comment]
There is now an OpenSearch extension in the Quarkiverse: https://github.com/quarkiverse/quarkus-opensearch
It is still a WIP but a release should be done soon with a first version.

**Comment by sorin-costea** [comment]
Just FYI I tried for my use case v0.2.4 and the high level rest client worked perfectly with AWS OpenSearch 2.3, thank you all!
